# HG changeset patch
# User pantxo <pantxo.diribarne@gmail.com>
# Date 1393532039 -3600
#      Thu Feb 27 21:13:59 2014 +0100
# Branch stable
# Node ID 3e53eca42f00651f7978f1d09fe088d9d01c3938
# Parent  51a0d1c1023c085862b7d892553ccfab7eafb2a1
provide better default for Mac keyboard layouts (bug #41337)

* ExtendedDefaultTranslatorMac.h: New file.
* KeyboardTranslator.cpp: Conditionally include it.

diff --git a/libgui/qterminal/libqterminal/unix/ExtendedDefaultTranslatorMac.h b/libgui/qterminal/libqterminal/unix/ExtendedDefaultTranslatorMac.h
new file mode 100644
--- /dev/null
+++ b/libgui/qterminal/libqterminal/unix/ExtendedDefaultTranslatorMac.h
@@ -0,0 +1,53 @@
+"keyboard \"Default (XFree 4)\""
+"key Escape : \"\\x1b\""
+"key Tab -Shift : \"\\t\"\n"
+"key Tab +Shift+Ansi : \"\\E[Z\"\n"
+"key Tab +Shift-Ansi : \"\\t\"\n"
+"key Backtab +Ansi : \"\\E[Z\"\n"
+"key Backtab -Ansi : \"\\t\"\n"
+"key Return-Shift-NewLine : \"\\r\"\n"
+"key Return-Shift+NewLine : \"\\r\\n\"\n"
+"key Return+Shift : \"\\EOM\"\n"
+"key Backspace : \"\\x7f\"\n"
+"key Meta +C: \"\\x03\"\n"
+"key Up -Shift+Ansi-AppCuKeys : \"\\E[A\"\n"
+"key Down -Shift+Ansi-AppCuKeys : \"\\E[B\"\n"
+"key Right-Shift+Ansi-AppCuKeys : \"\\E[C\"\n"
+"key Left -Shift+Ansi-AppCuKeys : \"\\E[D\"\n"
+"key Up -Ansi : \"\\E[1;*A\"\n"
+"key Down -Ansi : \"\\E[1;*B\"\n"
+"key Right -Ansi : \"\\E[1;*C\"\n"
+"key Left -Ansi : \"\\E[1;*D\"\n"
+"key Enter+NewLine : \"\\r\\n\"\n"
+"key Enter-NewLine : \"\\r\"\n"
+"key Home -AnyMod -AppCuKeys : \"\\E[H\"\n"
+"key End -AnyMod -AppCuKeys : \"\\E[F\"\n"
+"key Home -AnyMod +AppCuKeys : \"\\EOH\"\n"
+"key End -AnyMod +AppCuKeys : \"\\EOF\"\n"
+"key Home +AnyMod : \"\\E[1;*H\"\n"
+"key End +AnyMod : \"\\E[1;*F\"\n"
+"key Insert -AnyMod : \"\\E[2~\"\n"
+"key Delete -AnyMod : \"\\E[3~\"\n"
+"key Insert +AnyMod : \"\\E[2;*~\"\n"
+"key Delete +AnyMod : \"\\E[3;*~\"\n"
+"key Prior -Shift-AnyMod : \"\\E[5~\"\n"
+"key Next -Shift-AnyMod : \"\\E[6~\"\n"
+"key Prior -Shift+AnyMod : \"\\E[5;*~\"\n"
+"key Next -Shift+AnyMod : \"\\E[6;*~\"\n"
+"key F1 -AnyMod : \"\\EOP\"\n"
+"key F2 -AnyMod : \"\\EOQ\"\n"
+"key F3 -AnyMod : \"\\EOR\"\n"
+"key F4 -AnyMod : \"\\EOS\"\n"
+"key F5 -AnyMod : \"\\EOT\"\n"
+"key F6 -AnyMod : \"\\EOU\"\n"
+"key F7 -AnyMod : \"\\EOV\"\n"
+"key F8 -AnyMod : \"\\EOW\"\n"
+"key F9 -AnyMod : \"\\EOX\"\n"
+"key F10 -AnyMod : \"\\EOY\"\n"
+"key Space +Control : \"\\x00\"\n"
+"key Up +Shift-AppScreen : scrollLineUp\n"
+"key Prior +Shift-AppScreen : scrollPageUp\n"
+"key Down +Shift-AppScreen : scrollLineDown\n"
+"key Next +Shift-AppScreen : scrollPageDown\n"
+"key ScrollLock : scrollLock\n"
+"\0"
diff --git a/libgui/qterminal/libqterminal/unix/KeyboardTranslator.cpp b/libgui/qterminal/libqterminal/unix/KeyboardTranslator.cpp
--- a/libgui/qterminal/libqterminal/unix/KeyboardTranslator.cpp
+++ b/libgui/qterminal/libqterminal/unix/KeyboardTranslator.cpp
@@ -30,19 +30,27 @@
 
 // Qt
 #include <QtCore/QBuffer>
 #include <QtCore/QFile>
 #include <QtCore/QFileInfo>
 #include <QtCore>
 #include <QtGui>
 
+// FIXME: We should not have a special case for Mac here.  Instead, we
+// should be loading .keytab files at run time, and ideally, allowing
+// individual keys to be redefined from some preferences menu.
+
 //and this is default now translator - default.keytab from original Konsole
 const char* KeyboardTranslatorManager::defaultTranslatorText = 
+#if defined (Q_OS_MAC)
+#include "ExtendedDefaultTranslatorMac.h"
+#else
 #include "ExtendedDefaultTranslator.h"
+#endif
 ;
 
 KeyboardTranslatorManager::KeyboardTranslatorManager()
     : _haveLoadedAll(false)
 {
 }
 KeyboardTranslatorManager::~KeyboardTranslatorManager()
 {
